weapons/projectile.py

```diff
@@ -27,9 +26,5 @@ def update(self, model, lag_scalar):
         self.x += (self.move_x + self.momentum_x) * lag_scalar
         self.y += (self.move_y + self.momentum_y) * lag_scalar
         radius = self.collider.radius
-        if (self.impacted
-            or self.x < -radius
-            or self.x > (state.SCREEN_WIDTH + radius)
-            or self.y < -radius
-            or self.y > (state.SCREEN_HEIGHT + radius)):
+        if self.impacted:  # TODO: If level size is finite, remove projectile if it travels outside level.
```

Comment:
Currently we have a finite height but infinite width.  You might want to keep the checks for removing the object if it goes outside the height bounds.

Reply:
Oh, wait, I guess we also have an infinite y-axis as well.  Someone needs to make the background scroll in the Y direction too! :)

Reply:
yeah, not sure! I didnt want to do too much and add a level size. We could just scroll on the x -axis with the camera. I think the vertical scrolling is nice too.